src/RecordManager/Base/Command/Records/Export.php

```diff
@@ -716,4 +795,134 @@ protected function writeRecord(string $record): void
+                $pathPart['nodeName'],
+                null,
+                $this->getXmlNamespaceFromPrefix($xml, $pathPart['prefix'])
+            );
+            foreach ($pathPart['attrs'] as $attr => $value) {
+                $currentNode->addAttribute($attr, $value);
+            }
+        }
+        // See https://github.com/phpstan/phpstan/issues/8236
+        // @phpstan-ignore-next-line
+        $currentNode[0] = $nodeValue;
+    }
+
+    /**
+     * Try to find a node in XML element
+     *
+     * @param \SimpleXMLElement $xml      XML node
+     * @param array             $pathPart Path item
+     *
+     * @return ?\SimpleXMLElement
+     */
+    protected function findXmlNode(\SimpleXMLElement $xml, array $pathPart): ?\SimpleXMLElement
+    {
+        $ns = $this->getXmlNamespaceFromPrefix($xml, $pathPart['prefix']);
+        foreach ($xml->children($ns, true)->{$pathPart['nodeName']} as $candidate) {
+            if ($this->xmlNodeAttributesMatch($candidate, $pathPart['attrs'])) {
+                return $candidate;
+            }
+        }
+        return null;
+    }
+
+    /**
+     * Check if node attributes match the given array of attributes
+     *
+     * @param \SimpleXMLElement $node       Node
+     * @param array             $attributes Attributes to check
+     *
+     * @return bool
+     */
+    protected function xmlNodeAttributesMatch(\SimpleXMLElement $node, array $attributes): bool
+    {
+        foreach ($node->attributes() as $attr => $value) {
+            if (($attributes[$attr] ?? null) !== $value) {
+                return false;
+            }
+            unset($attributes[$attr]);
+        }
+        return !$attributes;
+    }
+
+    /**
+     * Get namespace from prefix
+     *
+     * @param \SimpleXMLElement $xml    Node
+     * @param ?string           $prefix Prefix
+     *
+     * @return string
+     */
+    protected function getXmlNamespaceFromPrefix(\SimpleXMLElement $xml, ?string $prefix): ?string
+    {
+        $ns = null;
+        $namespaces = $xml->getNamespaces();
+        if (null === $prefix) {
+            if (current($namespaces) !== 'http://www.w3.org/2001/XMLSchema-instance') {
+                $ns = current($namespaces);
+            }
+        } else {
+            $ns = $namespaces[$prefix] ?? $this->additionalNamespaces[$prefix] ?? null;
+        }
+        return $ns;
+    }
+
+    /**
+     * Parse XML path parameter
+     *
+     * @param ?string $path XPath-like path
+     *
+     * @return array
+     */
+    protected function parseXmlPath(?string $path): array
+    {
+        if (!$path) {
+            return [];
+        }
+        $result = [];
+        $pathParts = explode('/', $path);
+        foreach ($pathParts as $pathPart) {
+            // Extract node name and attributes:
+            $parts = explode('[', $pathPart);
+            $nodeParts = explode(':', $parts[0], 2);
+            if (isset($nodeParts[1])) {
```

Comment:
If you want to reduce number of lines of code, maybe:

```
if (count($nodeParts) < 2) {
    array_unshift($nodeParts, null);
}
[$prefix, $nodeName] = $nodeParts;
```

(Not that there's anything wrong with the other way...)

Reply:
I decided to ignore this to avoid `count` and array manipulation for imagined performance advantage. :)